mitosheet/mitosheet/step_performers/step_performer.py

```diff
@@ -82,6 +91,101 @@ def transpile(
         """
         pass
 
+    @classmethod
+    def execute_through_transpile(
+        cls,
+        prev_state: State,
+        params: Dict[str, Any],
+        execution_data: Optional[Dict[str, Any]]=None,
+        new_dataframe_params: Optional[ExecuteThroughTranspileNewDataframeParams]=None,
+        renamed_column_headers: bool=False,
+        new_column_params: Optional[ExecuteThroughTranspileNewColumnParams]=None,
+        use_deprecated_id_algorithm: Optional[bool]=False
+    ) -> Tuple[State, Dict[str, Any]]:
+        """
+        Some steps can be executed through the transpiled code -- and in these cases, we can call this function
+        so that we don't have to duplicate the work we do
+        """
+        if execution_data is None:
+            execution_data = {}
+
+        modified_dataframe_indexes = cls.get_modified_dataframe_indexes(params)
+        # If the modified indexes are -1, then only new dataframes have been created -- and in this
+        # case we just don't detect modifications
+        if modified_dataframe_indexes == {-1}:
+            modified_dataframe_indexes = set()
+
+        post_state = prev_state.copy(deep_sheet_indexes=modified_dataframe_indexes)
+
+        code_chunks = cls.transpile(post_state, params, execution_data)
+        code = []
+        for chunk in code_chunks:
+            _code, imports = chunk.get_code()
+            code.extend(imports)
+            code.extend(_code)
+
+        final_code = "\n".join(code)
+
+        # NOTE; this is all wrong, but it's just b/c it's hard to get things out of exec. We do our best
+        exec_globals = get_globals_for_exec(post_state, post_state.public_interface_version)
+        exec_locals = {**exec_globals}
```

Comment:
I guess this is what your comment is referring to here, but I don't get why we're exceccing the globals and then calling that exec_locals and then later passing both exec_globals and exec_locals to the exec function?